cephadm-preflight.yml

```diff
@@ -0,0 +1,89 @@
+---
+# This playbook configures the Ceph repository.
+# It also installs some prerequisites (podman, lvm2, chronyd, ...)
+#
+# Usage:
+#
+# ansible-playbook -i <inventory host file> cephadm-preflight.yml
+#
+# You can limit the execution to a set of hosts by using `--limit` option:
+#
+# ansible-playbook -i <inventory host file> cephadm-preflight.yml --limit <my_osd_group|my_node_name>
+#
+# You can override variables using `--extra-vars` parameter:
+#
+# ansible-playbook -i <inventory host file> cephadm-preflight.yml --extra-vars "ceph_origin=rhcs"
+#
+
+
+- hosts: all
+  become: true
+  gather_facts: true
+  vars:
+    ceph_origin: community
+    ceph_rhcs_version: 5
+    ceph_mirror: https://download.ceph.com
+    ceph_stable_key: https://download.ceph.com/keys/release.asc
+    ceph_release: pacific
+    upgrade_ceph_packages: false
+    ceph_pkgs:
+      - lvm2
+      - chrony
+      - cephadm
+      - podman
+      - python3
```

Comment:
```suggestion
```

need to set all defaults in a role

Reply:
In this case, these variables would need to be defined in `group_vars/all.yml` and not a role.

Reply:
I don't think having default values in a versioned file intended to be modified by users is very elegant.

Reply:
Ah thank you for clarifying. Sorry I misunderstood what you meant by `role` and made an (improper) assumption that you intended to set these variables in `./roles/<somewhere>` which would violate variable precedence from when this playbook is called. 

If the defaults are to remain in this file I agree that makes a lot of sense. I've had debugging pain points with how variables are shuffled around in ceph-ansible, though they had good reasons to do so.

Reply:
They would be defined in `./roles/ceph-defaults/defaults/main.yml`, yes.
This way you can override them in several places: inventory, group_vars/host_vars... and extra vars, if we keep them defined at the play vars level, it won't be possible (except at the extra vars level since it has the greatest precedence)